kong/admin_service.go

```diff
@@ -0,0 +1,174 @@
+package kong
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+)
+
+// AdminService handles Admins in Kong.
+type AdminService service
+
+// Invite creates an Admin in Kong.
+func (s *AdminService) Invite(ctx context.Context,
+	admin *Admin) (*Admin, error) {
+
+	if admin == nil {
+		return nil, errors.New("cannot create a nil admin")
+	}
+
+	endpoint := "/admins"
+	method := "POST"
+	req, err := s.client.NewRequest(method, endpoint, nil, admin)
+	if err != nil {
+		return nil, err
+	}
+
+	var createdAdmin struct {
+		Admin Admin `json:"admin,omitempty" yaml:"admin,omitempty"`
+	}
+	_, err = s.client.Do(ctx, req, &createdAdmin)
+	if err != nil {
+		return nil, err
+	}
+	return &createdAdmin.Admin, nil
+}
+
+// Create aliases the Invite function as it performs
+// essentially the same operation.
+func (s *AdminService) Create(ctx context.Context,
+	admin *Admin) (*Admin, error) {
+	return s.Invite(ctx, admin)
+}
+
+// Get fetches a Admin in Kong.
+func (s *AdminService) Get(ctx context.Context,
+	nameOrID *string, generateRegisterURL bool) (*Admin, error) {
```

Comment:
Can we split this into two functions?
Get(ctx, string): This one fetches the admin
GenerateRegisterURL(ctx,string): This one generates a register URL.